assemblyline/cachestore/__init__.py

```diff
@@ -23,7 +23,7 @@ def __init__(self, component: str, config=None, datastore=None):
 
         self.component = component
         self.datastore = datastore or forge.get_datastore(config=config)
-        self.filestore = FileStore(*config.filestore.cache)
+        self.filestore = FileStore(*config.filestore.cache, use_mi=config.filestore.use_mi)
```

Comment:
As this is a change that is only for the Azure Filestore, we should not create an extra params that will be passed to all filestores. You should get rid of the use_mi params everywhere and add it as a query param value to the filestore URL.

Reply:
Good Idea!!